packages/medusa/src/services/customer.js

```diff
@@ -416,6 +422,13 @@ class CustomerService extends BaseService {
         customer.password_hash = await this.hashPassword_(password)
       }
 
+      if (groups) {
+        const cgRepo = manager.getCustomRepository(
+          this.customerGroupRepository_
+        )
+        await this.updateCustomerGroups(customer, groups, cgRepo)
```

Comment:
Would prefer if we could get around having to loop and query each of the ids maybe we could do this:

```suggestion
        customer.groups = await this.customerGroupService_.list({ id: groups.map(({ id }) => id) })
```

To make sure that all the passed groups also exist we could then maybe do this:
```
if (customer.groups.length !== groups.length) {
  // Figure out what is missing and throw
}
```